aten/src/ATen/native/cpu/UpSampleMoreKernel.cpp

```diff
@@ -43,33 +107,73 @@ void cpu_upsample_nearest_backward(
   int64_t output_slice_size = output_depth * output_height * output_width;
   int64_t input_slice_size = input_depth * input_height * input_width;
 
+  using opmath_t = at::opmath_type<scalar_t>;
   auto loop1d = [&](int64_t begin, int64_t end) {
+    opmath_t* acc_data_ptr = nullptr;
+    std::unique_ptr<opmath_t[]> buffer_data;
+    if (std::is_same<scalar_t, BFloat16>::value) {
```

Comment:
I guess it's safer to check `opmath_t` and `scalar_t` doesn't match rather than explicitly checking bfloat16 here.

Reply:
Suggest to fix similar usage in other places too.

Reply:
Fix as suggested.